File: nestjs-api-gateway/src/modules/auth/auth.service.ts
import {  Injectable } from '@nestjs/common';
import { HttpService } from '@nestjs/axios';

import { ConfigService } from '@nestjs/config';

import { BaseApiService } from 'src/common/services/base-api.service';

@Injectable()
export class AuthService extends BaseApiService {
    constructor(httpService: HttpService, configService: ConfigService) {
        super(httpService, configService, 'AUTH_SERVICE_URL');
    }

    async loginAdmin(credentials: { username: string; password: string }) {
        const request$ = this.httpService.post(
            `${this.serviceUrl}/api/admin/auth/login`,
            credentials,
            {
                headers: { 'Content-Type': 'application/json' },
            }
        );

        return this.makeRequest(request$);
    }
    async loginCustomer(credentials: { username: string; password: string }) {
        const request$ = this.httpService.post(
            `${this.serviceUrl}/api/customer/auth/login`,
            credentials,
            {
                headers: { 'Content-Type': 'application/json' },
            }
        );

        return this.makeRequest(request$);
    }
    
}

File: java-auth-ms/src/main/java/dev/scastillo/auth/domain/service/AuthService.java
package dev.scastillo.auth.domain.service;

import dev.scastillo.auth.adapter.web.dto.UserInfoDto;
import dev.scastillo.auth.domain.service.dto.AuthLoginResponse;

public interface AuthService {
    AuthLoginResponse loginCustomer(String username, String password);
    AuthLoginResponse loginAdmin(String username, String password);
    UserInfoDto validateToken (String token);

}
